salt/client/__init__.py

```diff
@@ -863,9 +882,9 @@ def get_iter_returns(
         # iterator for this job's return
         if self.opts['order_masters']:
             # If we are a MoM, we need to gather expected minions from downstreams masters.
-            ret_iter = self.get_returns_no_block(jid, tags_regex=['^syndic/.*/{0}'.format(jid)])
+            ret_iter = self.get_returns_short_block(jid, tags_regex=['^syndic/.*/{0}'.format(jid)], wait=timeout / 4)
         else:
-            ret_iter = self.get_returns_no_block(jid)
+            ret_iter = self.get_returns_short_block(jid, wait=timeout / 4)
```

Comment:
This appears to be the line that's causing the test suite to hang. Setting the wait here back to `0.1` resolves the issue.